File: array/SecondLargest.java
public class SecondLargest {
    public static int getSecondLargest(int[] arr){
        int n = arr.length;
        int largest = Integer.MIN_VALUE;
        int secondLargest = Integer.MIN_VALUE;
        for(int i =0;i<n;i++){
            if(arr[i] > largest ){
                secondLargest = largest;
                largest = arr[i];
            }else if(arr[i] > secondLargest && arr[i] < largest){
                secondLargest = arr[i];
            }

        }
        return secondLargest;
    }
    public static void main(String[] args) {
        int arr[] = {5,2,7,9,4,12,8,0};
        System.out.println(getSecondLargest(arr));
    }
}

File: array/second_largest.js
function second_largest(arr){
    let largest = -1;
    let second_largest = -1;
    for(let i =0;i<arr.length;i++){
        if(arr[i] > largest){
            second_largest = largest;
            largest = arr[i];
        }else if(arr[i] > second_largest && arr[i] < largest){
            second_largest = arr[i];
        }
    }
    return second_largest;
}


let arr = [5,2,7,9,12,4,8,0];

console.log(second_largest(arr));